src/disp/syscall.c

```diff
@@ -1270,6 +1270,28 @@ static MVMDispSysCall async_unix_listen = {
     .expected_concrete = { 1, 1, 1, 1, 0 },
 };
 
+/* handle-open-mode */
+static void handle_open_mode_impl(MVMThreadContext *tc, MVMArgs arg_info) {
+    MVMOSHandle *oshandle  = (MVMOSHandle *)get_obj_arg(arg_info, 0);
+
+    MVMIOFileData *data = (MVMIOFileData *)oshandle->body.data;
```

Comment:
This is an unsafe pointer cast. We don't know that the handle is actually a file handle. At worst this can lead to a segfault.

I still say that rather than accessing the handle's data directly, the functionality should be hidden behind a proper IO op function pointer.

Reply:
@niner are you worried about a fuzzing injection vector?

Reply:
Ah, that makes total sense. Applied.